CS-FUNdamentals.rb

```diff
@@ -0,0 +1,54 @@
+
+def print_array(array)
+  print "["
+  i = 0
+  array.length.times do
+  if i == array.length-1
+  print array[i]
+  print  "]"
+  else
+  print array[i]
+  print ", "
+  i += 1
+  end
+  end
+end
+
+def stretch(array)
+  returnArray = []
+  array.each do |num|
+    i = num/2
+    modulo = num % 2
+    returnArray << (i + modulo)
+    returnArray << i
+  end
+  return returnArray
+end
+
+def numUnique(integers)
+  newHash = (integers.each_with_object(Hash.new(0)) { |integers, hash| hash[integers] += 1 }).keys.length
+  return newHash
+end
```

Comment:
Whoa - also the first time I've seen something like this. I think it's a bit hard to read as a single line, but it does what others solutions (that I showed in class) do over multiple lines. And as long as you are condensing to one line, why not just remove `newHash =` and return that one long line?